bftclient/src/matcher.cpp

```diff
@@ -15,7 +15,7 @@ namespace bft::client {
 
 std::optional<Match> Matcher::onReply(UnmatchedReply&& reply) {
   if (!valid(reply)) return std::nullopt;
-
+  if (config_.ignore_primary_) reply.metadata.primary = ReplicaId{0};
```

Comment:
I realize you are using the type of primary as is, but this is somewhat misleading.  ReplicaId{0} is completely arbitrary, and other code may assume that it actually matched as ReplicaId{0}. I don't think the value is used outside the matcher, but it may be worth changing the type of `reply.metadata.primary` to be a `std::optional<ReplicaId>`.

Reply:
fixed